src/Moryx.Resources.Management.Endpoints/ResourceManagementController.cs

```diff
@@ -0,0 +1,375 @@
+            _serialization = new ResourceSerialization();
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status417ExpectationFailed)]
+        [Route("types")]
+        public ActionResult<ResourceTypeModel> GetTypeTree()
+        {
+            var converter = new ResourceToModelConverter(_resourceTypeTree, _serialization);
+            return converter.ConvertType(_resourceTypeTree.RootType);
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status417ExpectationFailed)]
+        public ActionResult<ResourceModel[]> GetDetailsBatch([FromQuery] long[] ids)
+        {
+            var converter = new ResourceToModelConverter(_resourceTypeTree, _serialization);
+
+            if (ids is null)
+                ids = _resourceModification.GetResources<PublicResource>().Select(r => r.Id).ToArray();
+
+            return ids.Select(id => _resourceModification.Read(id, r => converter.GetDetails(r)))
+                .Where(details => details != null).ToArray();
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status417ExpectationFailed)]
+        [Route("query")]
+        public ActionResult<ResourceModel[]> GetResources(ResourceQuery query)
+        {
+            var filter = new ResourceQueryFilter(query, _resourceTypeTree);
+            var resourceProxies = _resourceModification.GetResources<PublicResource>(filter.Match).ToArray();
+
+            var converter = new ResourceQueryConverter(_resourceTypeTree, _serialization, query);
+            return resourceProxies.Select(p => _resourceModification.Read(p.Id, r => converter.QueryConversion(r))).ToArray();
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status417ExpectationFailed)]
+        [Route("{id}")]
+        public ActionResult<ResourceModel> GetDetails(long id)
+        {
+            var converter = new ResourceToModelConverter(_resourceTypeTree, _serialization);
+            var resourceModel = _resourceModification.Read(id, r => converter.GetDetails(r));
+            if (resourceModel is null)
+                return NotFound($"Resource '{id}' not found!");
+
+            return resourceModel;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status417ExpectationFailed)]
+        [Route("{id}/invoke/{method}")]
+        public ActionResult<Entry> InvokeMethod(long id, string method, Entry parameters)
+        {
+            if (_resourceModification.GetResource<PublicResource>(id) is null)
```

Comment:
This will never return a value, since it only allows for interfaces.

Reply:
means i need to use IPublicResource?